vm/src/builtins/range.rs

```diff
@@ -634,10 +634,14 @@ impl TryFromObject for RangeIndex {
         match_class!(match obj {
             i @ PyInt => Ok(RangeIndex::Int(i)),
             s @ PySlice => Ok(RangeIndex::Slice(s)),
-            obj => Err(vm.new_type_error(format!(
-                "sequence indices be integers or slices, not '{}'",
-                obj.class().name,
-            ))),
+            obj => match vm.to_index(&obj) {
+                Ok(val) => Ok(RangeIndex::Int(val)),
+                Err(_) => Err(vm.new_type_error(format!(
+                    "sequence indices be integers or slices or classes that override __index__ operator, not '{}'",
+                    obj.class().name
+                ))),
+            },
         })
     }
 }
+ 
```

Comment:
this line look like unnessessary change

Reply:
this line also causes lint error: https://github.com/RustPython/RustPython/pull/2807/checks?check_run_id=3273128526

Reply:
That line was added by rustfmt command but I also think it is unnecessary. I'll pick it out with below conversation.

Reply:
I'm sorry. I misunderstood that change. I resend v3 code and lint seems good now.